beginner_source/export_simple_model_to_onnx_tutorial.py

```diff
@@ -0,0 +1,150 @@
+# 2. Export the model to ONNX format.
+# 3. Save the ONNX model in a file.
+# 4. Visualize the ONNX model graph using `Netron <https://github.com/lutzroeder/netron>`_.
+# 5. Execute the ONNX model with `ONNX Runtime
+#
+# Note that because the ONNX exporter uses ``onnx`` and ``onnxscript`` to translate PyTorch operators into ONNX operators,
+# we will need to install them.
+# %%
+#  .. code-block:: bash
+#
+#      %%bash
+#      pip install onnx
+#      pip install onnxscript
+#
+# Once your environment is set up, let’s start modeling our image classifier with PyTorch,
+# exactly like we did in the 60 Minute Blitz tutorial.
+#
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class Net(nn.Module):
+
+    def __init__(self):
+        super(Net, self).__init__()
+        self.conv1 = nn.Conv2d(1, 6, 5)
+        self.conv2 = nn.Conv2d(6, 16, 5)
+        self.fc1 = nn.Linear(16 * 5 * 5, 120)
+        self.fc2 = nn.Linear(120, 84)
+        self.fc3 = nn.Linear(84, 10)
+
+    def forward(self, x):
+        x = F.max_pool2d(F.relu(self.conv1(x)), (2, 2))
+        x = F.max_pool2d(F.relu(self.conv2(x)), 2)
+        x = torch.flatten(x, 1)
+        x = F.relu(self.fc1(x))
+        x = F.relu(self.fc2(x))
+        x = self.fc3(x)
+        return x
+
+net = Net()
+
+# Analogous to the 60 Minute Blitz tutorial, we need to create a random 32x32 input.
+
+input = torch.randn(1, 1, 32, 32)
+
+# That is all we need to export the model to ONNX format: a model instance and a dummy input.
+# We can now export the model with the following code:
+
+export_output = torch.onnx.dynamo_export(net, input)
+
+# As we can see, we didn't need any code change on our model.
+# The resulting ONNX model is saved within ``torch.onnx.ExportOutput`` as a binary protobuf file.
+#
+# We can save it to disk with the following code:
+
+export_output.save("my_image_classifier.onnx")
+
+# Now that we have our model saved, we can visualize it with `Netron <https://github.com/lutzroeder/netron>`_.
+# Netron can either be installed on MacOS, Linux or Windows computers, or run directly from the browser.
+# Let's try the web version by opening the following link: https://netron.app/.
+#
+# .. image:: ../_static/img/onnx/netron_web_ui.png
+#
+# Once Netron is open, we can drag and drop our ``my_image_classifier.onnx`` file into the browser or select it after
+# clicking on `Open model` button.
+#
+# .. image:: ../_static/img/onnx/image_clossifier_onnx_modelon_netron_web_ui.png
+#
+# And that is it! We have successfully exported our PyTorch model to ONNX format and visualized it with Netron.
+#
+# The last step is executing the ONNX model with `ONNX Runtime`, but before we do that, let's install ONNX Runtime.
+# %%
+#  .. code-block:: bash
+#
+#      %%bash
+#      pip install onnxruntime
+
+# One aspect that wasn't mentioned before was that the exported ONNX Model may have more inputs than the original PyTorch model.
```

Comment:
Not clear what this is trying to say, I would just leave this out and discuss this point in an advanced tutorial